### ISSUE TITLE
No HTML5 support

### ISSUE THREAD

**Issue by Samistine** [SKCraft/Launcher#144]
This isn't working for me:
https://samistine.com/modloader/news

**Comment by sk89q** [comment]
Unfortunately the "browser" in Java is very bad and there's no straightforward alternative. Java 8+ comes with a proper browser (WebKit) but that would force users to need Java 8 to run the launcher.

If/when I make a new version of the launcher, I'll prob switch out of Java entirely (Electron maybe?).

**Comment by spannerman79** [comment]
I hate to suggest it but perhaps moving to Java 8 isn't a bad idea @sk89q 

Java 7 become EOL after April 2015 [Source](https://java.com/en/download/faq/java_7.xml) unless the user (highly unlikley) has a support contact with Oracle.

**Comment by sk89q** [comment]
The main problem with moving to Java 8 is that a lot of users unknowingly have an old version of Java, and the launcher simply won't run and won't even emit a useful error message to the user.

With Windows, the problem can somewhat be rectified by wrapping the .jar in an .exe but I'm not sure what the situation is with Mac OS X.

**Comment by wizjany** [comment]
Also despite Java 7 being EOL, the number of users who still use it is possibly as high as 30%.

**Comment by Samistine** [comment]
Is it possible to use java 8 features only when being run by a java 8 jre?

**Comment by sk89q** [comment]
Yes, in this particular case. I've considered it but I haven't gotten around to it.

I also haven't actually used the new WebKit browser so I don't know how well it works or how easily it integrates into the UI framework the launcher is currently using.

**Comment by James137137** [comment]
Electron! sounds interesting

**Comment by dscso** [comment]
I think Electron is a nice idea but I think the biggest problem is that Electron is a normal binary. Electron is based on Chromium and Chromium is **REALY BIG**. You can't compile Chromium with 32 Bit because 32 Bit computer havn't enough RAM 😂.

**Comment by James137137** [comment]
I tried added html5 support to my launcher (forked for this project), about 2 months ago I removed it because I wasn't happy with how much ram it was using plus it required java 8

**Comment by sk89q** [comment]
Well most people have broadband now, so it's not *that* big. I'm more worried about memory usage, but in comparison, Java uses a lot of memory as well.

The main plus of Electron is (1) making the launcher look good (like [the Battle.net launcher](https://i.imgur.com/p21bHF5.png)) is extremely easy and (2) it's an active project with a lot of people doing the same thing. There are other options (i.e. C++ and Qt) if you you want to use HTML/CSS UIs but you end up embedding Chrome again anyway, though at least things like the log window don't have to be in Chrome..

**Comment by jtrent238** [comment]
# I Agree the Battle.net Launcher would be amazing!

![](https://i.imgur.com/p21bHF5.png)

**Comment by jtrent238** [comment]
#163
